### ISSUE TITLE
IoTHub Connection Strings

### ISSUE THREAD

**Issue by szaroubi** [terraform-providers/terraform-provider-azurerm#1743]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
After provisioning an IoT Hub and an Azure Function, I would like to set a Function App setting (app_setting) with one of the IoTHub connection strings.

In order to accomplish this, one can generate the connection strings according to this document.
https://blogs.msdn.microsoft.com/iotdev/2017/05/09/understand-different-connection-strings-in-azure-iot-hub/

### New or Affected Resource(s)
Not sure, don't know the codebase enough.

### Potential Terraform Configuration

```hcl
resource "azurerm_function_app" "main" {
  name                      = "iot-function-test"
  .
  .
  .
  app_settings{
    IoTHubConnectionString = "${azurerm_iothub.main.eventhubconnectionstring}
  }
}
```

**Comment by szaroubi** [comment]
Hello @tombuildsstuff ,
Thank you for the very prompt reply.

Seeing I am a total n00b when it comes to building and running the terraform-provier-azurerm project.
Could I ask you to point me to some doc on how to set up my env ?

A part from the README.md file, is there other documentation on how to setup and test this provider (or other providers) with an existing terraform build (or generate a new terraform build)?

**Comment by szaroubi** [comment]
Here is my PR: https://github.com/terraform-providers/terraform-provider-azurerm/pull/1789

**Comment by tombuildsstuff** [comment]
hey @szaroubi 

Thanks for that PR - that's now been merged and will go out as part of the next release of the AzureRM Provider :)

Thanks!

**Comment by tombuildsstuff** [comment]
hey @szaroubi

Just to let you know that this has been released in v1.14.0 of the AzureRM Provider which is now available: https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.14.0/CHANGELOG.md

Thanks!
